### PR TITLE
Overlay

### PR DESCRIPTION
**Pull request by chriskr**

### CODE REVIEW

File: src/network/network_view.js [hunk 1]
```diff
@@ -917,3 +926,38 @@ cls.NetworkLog.create_ui_widgets = function()
   messages.addListener("view-created", on_view_created);
   messages.addListener("view-destroyed", on_view_destroyed);
 }
+
+cls.DetailOverlayView = function(id, container_class, html, default_handler, service)
+{
+  this.createView = function(container)
+  {
+    container.innerHTML = "<h2>hello</h2>";
```
Comment (hzr) [inline src/network/network_view.js:934]: Looks like temporary code.

File: src/network/network_view.js [hunk 2]
```diff
@@ -917,3 +926,38 @@ cls.NetworkLog.create_ui_widgets = function()
...
+  };
+
+  // overlay view has no name
+  this.init(id, container_class, html, default_handler);
+
```
Comment (hzr) [inline src/network/network_view.js]: Extra newline.

File: src/ui-scripts/celloverlay.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+﻿var CellOverlay = function(view_id, parent_cell, parent_view_id)
+{
+  this._init(view_id, parent_cell, parent_view_id);
+};
+
+CellOverlayPrototype = function()
```
Comment (hzr) [inline src/ui-scripts/celloverlay.js]: `var`.

File: src/network/network_view.js (hunk 1 above)
Comment (chriskr) [inline src/network/network_view.js:934]: Yes, this is just a test implementation of an actual overlay view. Daniel will have to do the real thing.

File: src/ui-scripts/cells.js [hunk 4]
```diff
@@ -244,12 +288,17 @@
         rough_cell.height = stored_height;
     }
 
+    this.min_width = rough_cell.min_width != null?
```
Comment (hzr) [inline src/ui-scripts/cells.js]: Should have a space before the `?`. Might format it with the `?` on the next line too.

File: src/ui-scripts/cells.js [hunk 5]
```diff
@@ -628,7 +694,7 @@
   {
     var delta_applied = 0;
     var child = null, i = 0;
-    var deltas = [];
+    var deltas = [], min = 0, max = 0;
```
Comment (hzr) [inline src/ui-scripts/cells.js]: One declaration per line.

File: src/ui-scripts/cells.js [hunk 6]
```diff
@@ -672,16 +738,17 @@
     }
     else
     {
+      min = this['min_' + dim];
       if( delta)
       {
         var newDim = this[dim] + delta;
-        if( newDim >= defaults['min_view_' + dim] || newDim >= this[dim] )
+        if( newDim >= min || newDim >= this[dim] )
```
Comment (hzr) [inline src/ui-scripts/cells.js]: Might as well fix the whitespace.

File: src/ui-scripts/overlaybackground.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+  this.init = function(cell)
+  {
+    this.cell = cell;
+    this.initBase();
+  };
+};
+
+OverlayBackgroundPrototype.prototype = UIBase;
+OverlayBackground.prototype = new OverlayBackgroundPrototype();
```
Comment (hzr) [inline src/ui-scripts/overlaybackground.js:81]: Quite a few whitespace issues in this file.

File: src/ui-scripts/overlayview.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    if (cell)
+      cell.hide_overlay(this.id);
+    this.is_active = false;
+  };
+
+  this.init = function(id, container_class, html, default_handler)
+  {
+    ViewBase.init.call(this, id, "", container_class, html, default_handler);
+  };
+};
```
Comment (hzr) [inline src/ui-scripts/overlayview.js:25]: Newline before this.

File: src/ui-scripts/ui-actions.js [hunk 9]
```diff
@@ -306,6 +306,15 @@ eventHandlers.click["toolbar-single-select"] = function(event, target)
   }
 }
 
+eventHandlers.click["close-overlay-view"] = function(event, target)
```
Comment (hzr) [inline src/ui-scripts/ui-actions.js]: `window.event_handlers`.

File: src/ui-style/ui.css [hunk 10]
```diff
@@ -1835,3 +1840,10 @@ info {
   overflow: hidden;
   text-overflow: ellipsis;
 }
+
+.background-overlay
+{
+  position: absolute;
+  border: 1px solid #999999;
```
Comment (hzr) [inline src/ui-style/ui.css]: 999

File: src/ui-style/ui.css [hunk 11]
```diff
@@ -1835,3 +1840,10 @@ info {
...
+  box-shadow: 0 2px 5px hsla(0, 0%, 0%, .5);
```
Comment (hzr) [inline src/ui-style/ui.css]: 0.5 according to the style guide.

File: src/ui-scripts/cells.js (hunk 4 above)
Comment (chriskr) [inline src/ui-scripts/cells.js]: It's actually Mathieu's code, but i can fix.

File: src/ui-scripts/overlaybackground.js (hunk 7 above)
Comment (chriskr) [inline src/ui-scripts/overlaybackground.js:81]: Yea, it was a bit a lot of copy (old code) and edit, will fix it.

**Comment by chriskr** [comment]
Pushed review fixes.

File: src/ui-strings/ui_strings-en.js [hunk 12]
```diff
@@ -600,6 +600,9 @@ ui_strings.S_CLEAR_NETWORK_LOG = "Clear network log";
 /* DESC: Close command line window */
 ui_strings.S_CLOSE_COMMAND_LINE = "Close console";
 
+/* DESC: Button label to close an overlay. */
+ui_strings.S_CLOSE_OVERLAY_VIEW = "Close overlay";
```
Comment (danfooo) [inline src/ui-strings/ui_strings-en.js:604]: I think I'd not speak of an "overlay". Maybe just "Close".

File: src/network/network_view.js [hunk 13]
```diff
@@ -425,6 +425,12 @@ cls.NetworkLogView = function(id, name, container_class, html, default_handler,
 
   this._on_clicked_request_bound = function(evt, target)
   {
+    if (this._overlay.is_active)
+      this._overlay.hide();
+    else
+      this._overlay.show();
+
+    return;
     var item_id = target.get_attr("parent-node-chain", "data-object-id");
```
Comment (p01) [inline src/network/network_view.js:434]: Dead code thanks to the `return;` above

File: src/network/network_view.js [hunk 14]
```diff
@@ -425,6 +425,12 @@ cls.NetworkLogView = function(id, name, container_class, html, default_handler,
 
   this._on_clicked_request_bound = function(evt, target)
   {
+    if (this._overlay.is_active)
+      this._overlay.hide();
+    else
+      this._overlay.show();
+
```
Comment (p01) [inline src/network/network_view.js:432]: Probably a common enough pattern to justifies a `show_hide()` method on the overlay class.

File: src/network/network_view.js (hunk 13 above)
Comment (chriskr) [inline src/network/network_view.js:434]: This is just to test the overlay, Daniel will have to finish it.

File: src/ui-strings/ui_strings-en.js (hunk 12 above)
Comment (chriskr) [inline src/ui-strings/ui_strings-en.js:604]: OK.

File: src/network/network_view.js (hunk 14 above)
Comment (chriskr) [inline src/network/network_view.js:432]: This is really just for testing, it will gone as soon.

**Pull request merged**